test/acceptance/PushPullDBInsertedNotificationsSpec.scala

```diff
@@ -137,10 +114,10 @@ class PushPullDBInsertedNotificationsSpec extends AcceptanceTestSpec
     result.status shouldBe ACCEPTED
   }
 
-  private val makeAPICall: ExpectedCall => Unit = expectedCall => callAPIAndMakeSureItReturns202(fakeRequestToAPI(expectedCall))
+  private val makeLocalCall: ExpectedCall => Unit = expectedCall => callAPIAndMakeSureItReturns202(fakeRequestToAPI(expectedCall))
   private val makeRemoteAPICall: ExpectedCall => Unit = expectedCall => callRemoteAPIAndMakeSureItReturns202(fakeRequestToAPI(expectedCall))
 
-  private def insertIntoDB: ExpectedCall => Unit = { expectedCall =>
+  private def insertToDB: ExpectedCall => Unit = { expectedCall =>
     val headers: Seq[Header] = expectedCall.maybeBadgeId.fold(Seq[Header]())(badgeId => Seq[Header](Header(X_BADGE_ID_HEADER_NAME, badgeId)))
 
```

Comment:
I think if you issue an `await(notificationRepo.save()...` here it may fix your DB insert out of sequence problem